### PR TITLE
chapter12: Reset Password

### PR DESCRIPTION
**Pull request by ngyihkai**
Ready

### CODE REVIEW

**Comment by ngyihkai** [comment]
Ready

File: app/assets/stylesheets/password_resets.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the PasswordResets controller here.
```
Comment (ghost) [inline app/assets/stylesheets/password_resets.scss]: delete this file if you don't use it

File: test/controllers/password_resets_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (ghost) [inline test/controllers/password_resets_controller_test.rb]: delete this file please

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
+  get 'password_resets/edit'
```
Comment (ghost) [inline config/routes.rb]: remove lines 2, 3, please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ngyihkai** [comment]
Ready

File: app/controllers/password_resets_controller.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def check_expiration
+    return unless @user.password_reset_expired?
+      flash[:danger] = t(:password_expired)
```
Comment (ngocvt-0484) [inline app/controllers/password_resets_controller.rb]: wrong indent
it should align with return

File: app/controllers/password_resets_controller.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def get_user
+    @user = User.find_by email: params[:email]
+  end
+
+  def valid_user
+    return if (@user && @user.activated && @user.authenticated?(:reset, params[:id]))
+      redirect_to home_url
```
Comment (ngocvt-0484) [inline app/controllers/password_resets_controller.rb]: wrong indent
it should align with return

File: app/helpers/password_resets_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module PasswordResetsHelper
```
Comment (ngocvt-0484) [inline app/helpers/password_resets_helper.rb]: delete this file because not use

**Comment by ngyihkai** [comment]
Ready

File: app/mailers/user_mailer.rb [hunk 7]
```diff
@@ -15,7 +15,7 @@ def account_activation(user)
   #   en.user_mailer.password_reset.subject
   #
   def password_reset
-    @greeting = "Hi"
-    mail to: "to@example.org"
+    @user = user
+  mail to: user.email, subject: "Password reset"
```
Comment (ngocvt-0484) [inline app/mailers/user_mailer.rb]: wrong indent
it should align with @user

File: config/locales/en.yml [hunk 8]
```diff
@@ -50,3 +50,8 @@ en:
   message: "message"
   error_acc_unacti: "Account not activated. Check your email for the activation link."
   pls_chk_email: "Please check your email to activate your account."
+  forgot_pass: "Forgot password"
+  email_sent: "Email sent"
+  email_not_found: "Email is not found"
+  password_reset: "Password reset"
+
```
Comment (ngocvt-0484) [inline config/locales/en.yml]: remove blank line

**Comment by ngyihkai** [comment]
Ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**